backend/src/v5/middleware/dataConverter/inputs/teamspaces/projects/models/drawings.js

```diff
@@ -0,0 +1,63 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+const { checkQuotaIsSufficient, fileFilter } = require('./commons/revisions');
+const { createResponseCode, templates } = require('../../../../../../utils/responseCodes');
+const Yup = require('yup');
+const YupHelper = require('../../../../../../utils/helper/yup');
+const { fileUploads } = require('../../../../../../utils/config');
+const { isRevAndStatusCodeUnique } = require('../../../../../../models/revisions');
+const { respond } = require('../../../../../../utils/responder');
+const { singleFileUpload } = require('../../../../multer');
+const { statusCodes } = require('../../../../../../models/modelSettings.constants');
+const { validateMany } = require('../../../../../common');
+
+const Drawings = {};
+
+const ACCEPTED_DRAWING_EXT = ['.dwg', '.pdf'];
+
+const validateRevisionUpload = async (req, res, next) => {
+	const schemaBase = {
+		statusCode: Yup.string().oneOf(statusCodes.map(({ code }) => code)).required(),
+		revCode: YupHelper.validators.alphanumeric(Yup.string().required().min(1).max(10)
+			.strict(true)),
+		desc: YupHelper.types.strings.shortDescription,
+	};
+
+	const schema = Yup.object(schemaBase).noUnknown().required()
+		.test('check-status-rev-code-uniqueness', 'The combination of statusCode and revCode needs to be unique', ({ revCode, statusCode }) => {
+			if (revCode && statusCode) {
+				const { teamspace, drawing } = req.params;
+				return isRevAndStatusCodeUnique(teamspace, drawing, revCode, statusCode);
+			}
+
+			// If either prop is missing we dont want to throw the test error message
+			// thus we are returning true and let it be caught by required()
+			return true;
+		});
+
+	try {
+		req.body = await schema.validate(req.body);
+		await next();
+	} catch (err) {
+		respond(req, res, createResponseCode(templates.invalidArguments, err?.message));
+	}
+};
+
+Drawings.validateNewRevisionData = validateMany([singleFileUpload('file', fileFilter(ACCEPTED_DRAWING_EXT), fileUploads.drawingSizeLimit, true), checkQuotaIsSufficient, validateRevisionUpload]);
```

Comment:
Can we also have a middleware to use the `fileExtensionFromBuffer` function from `utils/helper/typecheck` to verify the buffer is indeed what the extension say it is?

We already do this for images. From what I can tell the library can understand dwgs

Reply:
I have tested 3 dwg models from

![image](https://github.com/user-attachments/assets/b6cfff79-7de9-4176-b2ca-3f3cc3d0e669)

and the ext I get is undefined. It works every time with pdfs